tests/unit/optimization/lbfgs/LBFGSMinimizer_unittest.cc

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) The Shogun Machine Learning Toolbox
+ * Written (W) 2015 Wu Lin
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ *
+ */
+#include <gtest/gtest.h>
+#include "LBFGSMinimizer_unittest.h"
+
+CMyCostFunction::CMyCostFunction()
+	:CFirstOrderCostFunction()
+{
+	init();
+}
+void CMyCostFunction::init()
+{
+	m_obj=NULL;
+}
+CMyCostFunction::~CMyCostFunction()
+{
+	SG_UNREF(m_obj);
+}
+void CMyCostFunction::set_target(CPiecewiseQuadraticObject *obj)
+{
+	if(obj!=m_obj)
+	{
+		SG_UNREF(m_obj);
+		m_obj=obj;
+		SG_REF(m_obj);
+	}
+}
+
+float64_t CMyCostFunction::get_cost()
+{
+	REQUIRE(m_obj,"object not set\n");
+	return m_obj->get_value();
+}
+
+CMap<TParameter*, SGVector<float64_t> >* CMyCostFunction::obtain_variable_reference()
+{
+	REQUIRE(m_obj,"object not set\n");
```

Comment:
It seems a bit of an overkill to implement objective functions in this way....don't you think?